lib/puppet/defaults.rb

```diff
@@ -225,7 +229,7 @@ def self.default_vendormoduledir
     }
   )
 
-  define_settings(:main,
+  settings.define_settings(:main,
```

Comment:
What's the purpose of calling `define_settings` so many times? I think @Magisus suggested that it's just a remnant of when different settings were added? Perhaps we could condense these?

Reply:
`define_settings` adds a new definition for a settings object; these are all set before puppet reads from its own conf files, so puppet knows what to do given a setting. The class definition takes a section and then hash of definitions. The most common section is `:main`, repeated about 6 times? I wanted to isolate the scope of this change because it wasn't clear to me if there is anything that relies on the old behavior of what happens by simply requiring the `defaults.rb` file.

Reply:
So maybe we could do a follow PR that changes the indentation and collapses this into one hash per section?

Reply:
it's fine if you want to reformat this. I'd suggest making one commit that is just white space changes, so `git show -w` should show no changes. Then another commit to move stuff around. But if you do that then it will cause merge up conflicts from older branches. So you might want a separate PR and target 5.5.x.